lxd/auth/driver_rbac.go

```diff
@@ -117,7 +118,7 @@ func (r *rbac) load() error {
 	// Perform full sync when online
 	go func() {
 		for {
-			err = r.syncProjects()
+			err = r.syncProjects(r.ctx)
```

Comment:
We have a StopService() function, I wonder if we should also have a StartService() function, as Go recommends against having long-lived contexts inside a struct like the `load()` function suggests it does (I realise it doesn't actually do that though).

Also, having a StartService (that performs initial sync/activation) could return an error and avoid potentially LXD starting up in an inconsistent state with the authorizer service.

Reply:
rather than `load` accepting a context.

Reply:
I can see its retrying so that is nice too. Lets leave it as it is for now, but I think we should log an error at least if the sync process is failing.

Reply:
It does beg the question, if we've not successfully synced yet, should we be fulfilling requests using this authorizer when the data its using for the check is potentially out of date? What do you think, should an authorizer have a "in sync" status that affects how it responds?

Reply:
For the RBAC driver, the `UserAccess` it retrieved when a user authenticates. This forces a sync with the RBAC server and errors if the user is still not present in the local cache afterwards. The RBAC server also tells us if there has been a change on its end so we can resync everything.

Ultimately I didn't want to change the RBAC driver too much from what is existing for fear of breaking it.